### DISCUSSION TITLE
thoughts on terraform

### DISCUSSION THREAD

**Discussion by williamohara** [ory/kratos#2599] [Ideas]
does anyone have any thoughts on deploying fully via terraform? (i.e. no helm) 

Maybe even going so far as to developing a provider that will configure kratos via terraform? It looks like someone did id for Hydra... https://registry.terraform.io/providers/hypnoglow/oryhydra/latest/docs/resources/oauth2_client

**Comment by vinckr** [comment]
For Ory Hydra I would actually recommend this excellent Terraform provider:

https://github.com/svrakitin/terraform-provider-hydra
https://registry.terraform.io/providers/svrakitin/hydra/latest

I am not aware of a Terraform provider for Kratos - but I think it would be a great addition to our docs & [community projects](https://www.ory.sh/docs/ecosystem/community) as well :-)

**Reply by williamohara** [reply]
what about transcribing the helm templates into a full on terraform project? (still taking in a config yaml for kratos until a provider is done).The terraform provider for k8 is a lot better lately, although there is still a few shortcomings. but it gives me infrastructure setup and component setup in the same language. it gives me a better sense of what is happening too.

**Reply by vinckr** [reply]
I am not very familiar with Terraform myself, but if that is possible I think it would be a great addition. Open to contributions and I am happy to help along and learn more Terraform in the process.
